### PR TITLE
Clean up contact history

### PR DESCRIPTION
**Pull request by rowanseymour**
* Makes discerning item types more explicit by wrapping items, e.g. `if item.type == 'broadcast'` instead of `if item.recipient_count`
* Fixes several cases of variable shadowing

### CODE REVIEW

File: temba/contacts/templatetags/contacts.py [hunk 1]
```diff
@@ -78,9 +79,9 @@ def format_urn(urn, org):
 
 @register.filter
 def urn(contact, org):
-    urn = contact.get_urn()
```
Comment (rowanseymour) [inline temba/contacts/templatetags/contacts.py]: Variable shadows its own function name

File: temba/contacts/templatetags/contacts.py [hunk 2]
```diff
@@ -154,62 +154,56 @@ def extension(url):  # pragma: needs cover
...
 @register.filter
 def history_class(item):
-    css = ''
-    from temba.msgs.models import Msg, ERRORED, FAILED
-    from temba.ivr.models import IVRCall
-
-    if isinstance(item, Msg):
-        if item.media and item.media[:6] == 'video:':
-            css = '%s %s' % (css, 'video')
-        if item.direction or item.recipient_count:
```
Comment (rowanseymour) [inline temba/contacts/templatetags/contacts.py]: `Msg.direction` is always truthy and `recipient_count` is an attribute on `Broadcast`, not `Msg`

File: temba/contacts/templatetags/contacts.py [hunk 3]
```diff
@@ -97,26 +98,23 @@ def urn_icon(urn):
...
-        return location
+    (content_type, delim, loc) = geo_url.partition(':')
+    if len(loc.split(',')) == 2:
+        return loc
 
 
 @register.filter
 def media_url(media):
     if media:
-        # TODO: remove after migration msgs.0053
```
Comment (rowanseymour) [inline temba/contacts/templatetags/contacts.py]: Have checked and this was addressed

**Comment by nicpottier** [comment]
Can I expand your scope since you are in here and ask you insert webhook events that still for the contact / run? That would really help the resistbot peeps.

**Comment by rowanseymour** [comment]
@nicpottier on it but would like to do that in different PR so please take a gander at this one

File: templates/contacts/contact_history.haml [hunk 4]
```diff
@@ -153,34 +154,35 @@
                     .recording-text
                       %span.text
                         -if message.direction == 'O'
-                          -if item.text not in item.media
-                            {{item.text}}
+                          -if message.text not in item.media
```
Comment (norkans7) [inline templates/contacts/contact_history.haml]: `item.obj.media`

**Pull request merged**